source/opt/const_folding_rules.cpp

```diff
@@ -547,7 +547,7 @@ ConstantFoldingRule FoldOpDotWithConstants() {
     for (int i = 0; i < 2; ++i) {
       if (constants[i]) {
         if (constants[i]->AsNullConstant() ||
-            constants[i]->AsVectorConstant()->IsZero()) {
+            (constants[i]->AsVectorConstant()->IsZero())) {
```

Comment:
Needed?

Reply:
No, I was playing around with other stuff and I did not revert everything.